turbopy/core.py

```diff
@@ -214,10 +214,14 @@ def read_tools_from_input(self):
         if "Tools" in self.input_data:
             for tool_name, params in self.input_data["Tools"].items():
                 tool_class = ComputeTool.lookup(tool_name)
-                if len(params) >= 1:
-                    for tool in params:
-                        tool["type"] = tool_name
-                        self.compute_tools.append(tool_class(owner=self, input_data=tool))
+                if isinstance(params, list):
+                    if len(params) >= 1:
+                        for tool in params:
+                            tool["type"] = tool_name
+                            self.compute_tools.append(tool_class(owner=self, input_data=tool))
+                    else:
+                        params[0] = {"type": tool_name}
+                        self.compute_tools.append(tool_class(owner=self, input_data=params[0]))
                 else:
                     params["type"] = tool_name
                     self.compute_tools.append(tool_class(owner=self, input_data=params))
```

Comment:
This can be significantly simplified by making `params` a list if it isn't already, and then just looping through the list. Note that python has no trouble looping through a list with only one item.
```suggestion
                if not isinstance(params, list):
                    params = [params]
                for tool in params:
                    tool["type"] = tool_name
                    self.compute_tools.append(tool_class(owner=self, input_data=tool))
```

Reply:
When I had done that earlier, it raised issues of params with length 0, but now it seems to be working.

Reply:
Hmm... strange it didn't work before. But glad it's working now 🤷‍♂️